src/lib.rs

```diff
@@ -248,9 +277,63 @@ impl Bloom {
     #[classattr]
     const __hash__: Option<PyObject> = None;
 
-    /// Load from a file, see "Persistence" section in the README
+    /// Load from a file, fileobj or bytes, see "Persistence" section in the README
     #[classmethod]
-    fn load(_cls: &PyType, filepath: &str, hash_func: &PyAny) -> PyResult<Bloom> {
+    fn load(_cls: &PyType, src: PyObject, hash_func: &PyAny) -> PyResult<Bloom> {
+        match SaveLoadType::from_pyobject(Some(src)) {
+            Ok(f) => match f {
+                SaveLoadType::Filepath(filepath) => {
+                    let file = File::open(filepath)?;
+                    Self::load_from_read(hash_func, || Ok::<_, PyErr>(file))
+                }
+                SaveLoadType::FileLike(f) => Self::load_from_read(hash_func, || Ok::<_, PyErr>(f)),
+                SaveLoadType::Bytes(b) => Self::load_from_read(hash_func, || {
+                    Ok::<&[u8], std::convert::Infallible>(b.as_ref())
+                }),
+                _ => Err(PyErr::new::<PyTypeError, _>("Incorrect argument type")),
+            },
+            Err(e) => Err(e),
+        }
+    }
+
+    /// Save to a file, fileobj or bytes, see "Persistence" section in the README
+    fn save(&self, dest: Option<PyObject>) -> PyResult<PyObject> {
+        match SaveLoadType::from_pyobject(dest) {
+            Ok(f) => match f {
+                SaveLoadType::Filepath(filepath) => {
+                    let file = File::create(filepath)?;
+                    let _ = self.save_to_write(|| Ok::<_, PyErr>(file));
+                    Python::with_gil(|py| Ok(().to_object(py)))
+                }
+                SaveLoadType::FileLike(f) => {
+                    let _ = self.save_to_write(|| Ok::<_, PyErr>(f));
+                    Python::with_gil(|py| Ok(().to_object(py)))
+                }
+                SaveLoadType::Bytes(_) | SaveLoadType::None => {
+                    let mut bytes = Vec::with_capacity(
+                        mem::size_of_val(&self.k) + (self.filter.len() / 8) as usize,
+                    );
+                    self.save_to_write(|| Ok::<_, std::convert::Infallible>(&mut bytes))?;
+                    Python::with_gil(|py| Ok(PyBytes::new(py, &bytes).into()))
+                }
+            },
+            Err(e) => Err(e),
+        }
+    }
```

Comment:
This could become:

```rust
    fn save<'py>(&self, py: Python<'py>, dest: Option<SaveDest>) -> PyResult<Option<&'py PyBytes>> {
        Ok(match dest {
            Some(SaveDest::Filepath(filepath)) => {
                self.save_to_write(|| File::create(filepath))?;
                None
            }
            Some(SaveDest::FileLike(f)) => {
                self.save_to_write_obj(f)?;
                None
            }
            None => {
                let len = mem::size_of_val(&self.k) + (self.filter.len() / 8) as usize;

                // Write into a new PyBytes object directly
                let py_bytes = PyBytes::new_with(py, len, |bytes| {
                    let mut writer = &mut bytes[..];
                    self.save_to_write_obj(&mut writer)?;
                    // This should always be true if our len calculation was correct
                    assert!(writer.is_empty());
                    Ok(())
                })?;
                Some(py_bytes)
            }
        })
    }
```

I also used the `PyBytes::new_with` method to write directly into a PyBytes object, rather than writing to a Vec, then copying.